2.Zadaca/app/src/main/java/com/example/a2zadaca/InputFragment.kt

```diff
@@ -1,34 +1,129 @@
 package com.example.a2zadaca
 
 import android.os.Bundle
-import androidx.fragment.app.Fragment
+import android.text.InputType.TYPE_CLASS_NUMBER
 import android.view.LayoutInflater
 import android.view.View
+import android.view.View.GONE
+import android.view.View.VISIBLE
 import android.view.ViewGroup
+import android.widget.*
+import android.widget.AdapterView.OnItemSelectedListener
+import androidx.fragment.app.Fragment
 import androidx.lifecycle.ViewModelProvider
+import com.example.a2zadaca.data.Car
+import com.example.a2zadaca.data.EngineType
 import com.example.a2zadaca.databinding.FragmentInputBinding
 import com.example.a2zadaca.ui.main.PageViewModel
-import com.example.a2zadaca.ui.main.Person
 
 class InputFragment : Fragment(R.layout.fragment_input) {
     private lateinit var binding: FragmentInputBinding
     private lateinit var viewModel: PageViewModel
 
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
         binding = FragmentInputBinding.inflate(layoutInflater)
         viewModel = ViewModelProvider(requireActivity())[PageViewModel::class.java]
+        binding.engineSpinner.adapter = ArrayAdapter(
+            requireContext(),
+            android.R.layout.simple_spinner_dropdown_item,
+            EngineType.values()
+        )
+        when (binding.engineSpinner.selectedItem) {
+            EngineType.SELECT -> binding.radioGroup.visibility = GONE
+            EngineType.GASOLINE -> binding.radioGroup.visibility = VISIBLE
+            EngineType.DIESEL -> binding.radioGroup.visibility = VISIBLE
+            EngineType.HYBRID -> binding.radioGroup.visibility = VISIBLE
+            EngineType.ELECTRIC -> binding.radioGroup.visibility = GONE
+        }
```

Comment:
Mogao si napisati ovako:

```suggestion
binding.radioGroup.visibility = when (binding.engineSpinner.selectedItem) {
            EngineType.SELECT, EngineType.ELECTRIC -> GONE
            else -> VISIBLE
        }
```